amr-wind/utilities/tagging/TerrainElevation.H

```diff
@@ -0,0 +1,74 @@
+/*
+ * NOTE: This class is a work in progress.
+ * In the future, TerrainElevation may be extended to handle additional
+ * terrain-related data such as roughness, and could be used by other components
+ * of the codebase, including TerrainDrag, MetMastForcing, and similar modules.
+ */
+
+#ifndef TERRAINELEVATION_H_
+#define TERRAINELEVATION_H_
+
+#include "AMReX.H"
+#include <AMReX_Vector.H>
+#include <memory>
+#include <string>
+
+namespace amr_wind {
+
+class TerrainElevation
+{
+public:
+    TerrainElevation() = default;
+
+    void load_from_file(const std::string& filename);
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE bool is_loaded() const
+    {
+        return m_loaded;
+    }
+
+    static std::shared_ptr<TerrainElevation>& get_instance();
+    static void ensure_loaded(const std::string& filename);
+
+    // Accessors (const and non-const)
```

Comment:
This all feels like overkill. How about we just use a struct with public members? Also none of these should be `DEVICE` since we should not be accessing a Vector on device.